src/sagas.js

```diff
@@ -0,0 +1,126 @@
+// @flow
+
+import {
+  REQUEST_ONGOING_VERSIONS,
+  REQUEST_POLLBOT_VERSION,
+  UPDATE_URL,
+  REFRESH_STATUS,
+  REQUEST_STATUS,
+} from './types';
+import type {
+  APIVersionData,
+  CheckResults,
+  OngoingVersions,
+  ReleaseInfo,
+  RequestStatus,
+  State,
+} from './types';
+import {all, call, put, select, takeEvery} from 'redux-saga/effects';
+import {
+  checkStatus,
+  getOngoingVersions,
+  getPollbotVersion,
+  getReleaseInfo,
+} from './PollbotAPI';
+import {
+  addCheckResult,
+  localUrlFromVersion,
+  setVersion,
+  updateLatestChannelVersions,
+  updatePollbotVersion,
+  updateReleaseInfo,
+} from './actions';
+
+type Saga = Generator<*, void, *>;
+
+// Fetching the version from the Pollbot service.
+export function* fetchPollbotVersion(): Saga {
+  try {
+    const version: APIVersionData = yield call(getPollbotVersion);
+    yield put(updatePollbotVersion(version));
+  } catch (err) {
+    console.error('Failed getting the pollbot version', err);
+  }
+}
+
+// Fetching the ongoing versions.
+export function* fetchOngoingVersions(): Saga {
+  try {
+    const ongoingVersions: OngoingVersions = yield call(getOngoingVersions);
+    yield put(updateLatestChannelVersions(ongoingVersions));
+  } catch (err) {
+    console.error('Failed getting the latest channel versions', err);
+  }
+}
+
+// Update the url from the version stored in the state.
+export function* updateUrl(): Saga {
+  const state: State = yield select();
+  window.location.hash = localUrlFromVersion(state.version);
+}
+
+// Refreshing a status for the current version.
+export function* refreshStatus(): Saga {
+  const notifyChanges = (checkTitle, status) => {
+    if (Notification.permission === 'granted') {
+      new Notification(`${checkTitle}: status changed (${status}).`);
+    }
+  };
+
+  const state: State = yield select();
+  // Save previous results so we can check if something changed.
+  const prevResults = state.checkResults;
+  yield put(setVersion(state.version));
```

Comment:
Why do we need to set the version when we request a status? This might be outside the scope of this PR since I see we did it this way before..

Reply:
Yeah, it's not ideal, but it's the action that resets the check results (you don't want to be adding check results to the previous ones). I might modify this in the future to have a different action for that.